### ISSUE TITLE
[BUG] No bind variables have been provided with a single statement batch execution.

### ISSUE THREAD

**Issue by guillaume-alvarez** [opensearch-project/performance-analyzer#355]
**What is the bug?**
When using opensearch 2.3.0 or 2.4.1 docker the content of /usr/share/opensearch/logs/performance-analyzer.log is flooded with logs like
```
Dec 31, 2022 11:19:02 AM org.jooq.tools.JooqLogger info
INFO: Single batch             : No bind variables have been provided with a single statement batch execution. This may be due to accidental API misuse
```
7 new lines are printed every 5 seconds.

**How can one reproduce the bug?**
1. start opensearch 2.4.1
2. start a shell inside the docker
3. execute `tail -f  /usr/share/opensearch/logs/performance-analyzer.log`
4. also check space used by docker, it increases regularly

**What is the expected behavior?**
Either mask this log (if harmless) or fix the incorrect usage.

**What is your host/environment?**
 - OS: Windows+WSL2 or Ubuntu 22.10
 - Version 2.3.0 or 2.4.1
 - Plugins: none

**Do you have any screenshots?**
logs:
```
11:18:56.798 [rca-controller] ERROR org.opensearch.performanceanalyzer.rca.RcaController - Error reading file /usr/share/opensearch/data/rca_enabled.conf
java.nio.file.NoSuchFileException: /usr/share/opensearch/data/rca_enabled.conf
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218) ~[?:?]
	at java.nio.file.Files.newByteChannel(Files.java:380) ~[?:?]
	at java.nio.file.Files.newByteChannel(Files.java:432) ~[?:?]
	at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422) ~[?:?]
	at java.nio.file.Files.newInputStream(Files.java:160) ~[?:?]
	at java.util.Scanner.<init>(Scanner.java:718) ~[?:?]
	at org.opensearch.performanceanalyzer.rca.RcaController.lambda$readRcaEnabledFromConf$1(RcaController.java:344) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at org.opensearch.performanceanalyzer.core.Util.lambda$invokePrivileged$1(Util.java:57) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at java.security.AccessController.doPrivileged(AccessController.java:318) ~[?:?]
	at org.opensearch.performanceanalyzer.core.Util.invokePrivileged(Util.java:53) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at org.opensearch.performanceanalyzer.rca.RcaController.readRcaEnabledFromConf(RcaController.java:342) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at org.opensearch.performanceanalyzer.rca.RcaController.run(RcaController.java:300) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at org.opensearch.performanceanalyzer.PerformanceAnalyzerApp.lambda$startRcaTopLevelThread$0(PerformanceAnalyzerApp.java:173) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at org.opensearch.performanceanalyzer.threads.ThreadProvider.lambda$createThreadForRunnable$0(ThreadProvider.java:45) ~[performance-analyzer-rca-2.3.0.0.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
11:18:57.042 [pa-reader] ERROR org.opensearch.performanceanalyzer.reader.ReaderMetricsProcessor - Error reading file '/usr/share/opensearch/data/batch_metrics_enabled.conf': java.nio.file.NoSuchFileException: /usr/share/opensearch/data/batch_metrics_enabled.conf
Dec 31, 2022 11:18:57 AM org.jooq.tools.JooqLogger info
INFO: 
                                      
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@  @@        @@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@        @@@@@@@@@@
@@@@@@@@@@@@@@@@  @@  @@    @@@@@@@@@@
@@@@@@@@@@  @@@@  @@  @@    @@@@@@@@@@
@@@@@@@@@@        @@        @@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@        @@        @@@@@@@@@@
@@@@@@@@@@    @@  @@  @@@@  @@@@@@@@@@
@@@@@@@@@@    @@  @@  @@@@  @@@@@@@@@@
@@@@@@@@@@        @@  @  @  @@@@@@@@@@
@@@@@@@@@@        @@        @@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@  @@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@  Thank you for using jOOQ 3.10.8
                                      
11:18:57.402 [pa-reader] ERROR org.opensearch.performanceanalyzer.reader.ReaderMetricsProcessor - Error reading file '/usr/share/opensearch/data/batch_metrics_enabled.conf': java.nio.file.NoSuchFileException: /usr/share/opensearch/data/batch_metrics_enabled.conf
11:18:59.554 [pa-reader] ERROR org.opensearch.performanceanalyzer.reader.ReaderMetricsProcessor - Error reading file '/usr/share/opensearch/data/batch_metrics_enabled.conf': java.nio.file.NoSuchFileException: /usr/share/opensearch/data/batch_metrics_enabled.conf
Dec 31, 2022 11:19:02 AM org.jooq.tools.JooqLogger info
INFO: Single batch             : No bind variables have been provided with a single statement batch execution. This may be due to accidental API misuse
Dec 31, 2022 11:19:02 AM org.jooq.tools.JooqLogger info
INFO: Single batch             : No bind variables have been provided with a single statement batch execution. This may be due to accidental API misuse
Dec 31, 2022 11:19:02 AM org.jooq.tools.JooqLogger info
INFO: Single batch             : No bind variables have been provided with a single statement batch execution. This may be due to accidental API misuse
Dec 31, 2022 11:19:02 AM org.jooq.tools.JooqLogger info
INFO: Single batch             : No bind variables have been provided with a single statement batch execution. This may be due to accidental API misuse
Dec 31, 2022 11:19:02 AM org.jooq.tools.JooqLogger info
```

**Do you have any additional context?**
At the start of the logs there some errors about configuration files not set in default opensearch docker.
Not sure it is relevant? It does not prevent the performance-analyzer from working. I opened another issue for these: https://github.com/opensearch-project/opensearch-build/issues/3042

**Comment by guillaume-alvarez** [comment]
I also guess the JooqLogger does not use the log4j configuration, maybe that is the root cause?

**Comment by guillaume-alvarez** [comment]
**If the log is harmless** (which I don't know) then updating jooq to 3.12+ may help: https://github.com/jOOQ/jOOQ/issues/7643

**Comment by khushbr** [comment]
@guillaume-alvarez Can you confirm https://github.com/opensearch-project/performance-analyzer-rca/commit/a3e5a1bb7085dab23011d97dc6e8b50984b43f8d addresses your concern.

**Comment by guillaume-alvarez** [comment]
> @guillaume-alvarez Can you confirm [opensearch-project/performance-analyzer-rca@a3e5a1b](https://github.com/opensearch-project/performance-analyzer-rca/commit/a3e5a1bb7085dab23011d97dc6e8b50984b43f8d) addresses your concern.

I'm sorry but it does not.
The logs are no longer written in the same file but we still get megabytes of JOOQ errors.

**Comment by khushbr** [comment]
Fixed in https://github.com/opensearch-project/performance-analyzer-rca/commit/30ca0b302c3be7b36703b326772ea31cf065fded, this will be available as part of `v2.11.0` release.
